xclim/indices/_agro.py

```diff
@@ -1265,6 +1467,21 @@ def standardized_precipitation_evapotranspiration_index(
         Name of the fitting method, such as `ML` (maximum likelihood), `APP` (approximate). The approximate method
         uses a deterministic function that doesn't involve any optimization. Available methods
         vary with the distribution: 'gamma':{'APP', 'ML'}, 'fisk':{'ML'}
+    cal_start: DateStr | None
+        Start date of the calibration period. A `DateStr` is expected, that is a `str` in format `"YYYY-MM-DD"`. Default option `None`
+        means that the calibration period begins at the start of the input dataset.
+    cal_end: DateStr | None
+        End date of the calibration period. A `DateStr` is expected, that is a `str` in format `"YYYY-MM-DD"`. Default option `None`
+        means that the calibration period finishes at the end of the input dataset.
+    params: xarray.DataArray
+        Fit parameters. The `params` can be computed using ``xclim.indices.standardized_index_fit_params`` in advance.
+        The ouput can be given here as input, and it overrides other options.
+    offset: Quantified
```

Comment:
Suggestion:
```suggestion
    offset: Quantified | None = None,
```

```python3
# pseudo-code
if offset is None:
  if params is not None:
      if dist in bounded_distributions:
          offset = 1000 mm / d
      else: 
        offset = 0
    else:
        offset = params.offset
else:
    if params is not None and params.offset != offset:
        warning
```

C'est à dire : put a default of None and clearly explain in the docstring what the default behaviour is. I know I just suggested something different for the snow thing, but that was in a "temporary" perspective!

Reply:
Also, could there some option where `offset = -wb.min()` ?

Reply:
Currently, we have a legacy default "1 mm/d", I was trying not to breaking things for now. Ideally:
* By default, offset = None
* If a dist bounded by 0 I chosen but not offset is given, raise an error

I will check, but I think actually, there is not a restriction in scipy to use gamma/fisk distrubutions with 0 bounded data. A parameter can also be fitted for this shift. This is was an idea for another PR though.

the problem with `offset = -wb.min()` is that you can become sensitive to what data is included. Imagine your reference data is 1980-2010 and you fit two cases:
1. 1980-2020
2. 1980-2050

You could have a different minimum in cases 1 &  2, so different offset. Ideally, since this is a trick, we would like that our computation are not too sensitive on this. But in any, it would be nice that the results of 1. are en exact subset of 2. if you use the same calibration data / methods for both computations.

Reply:
Ideally, I would eventually like to get rid of the default 1 mm/day default. It was just copied on monocongo's climate indices. It's a good rule of thumb, but users have found cases where a bigger offset was needed. Maybe we can keep it in but warn users about having NaNs?  Maybe just in the docstring. Not sure. I could check the R implementation to see if I can dig ideas.